client/repolist.c

```diff
@@ -255,6 +255,10 @@ TDNFCreateRepoFromDirectory(
     BAIL_ON_TDNF_ERROR(dwError);
 
     dwError = TDNFIsDir(pszPath, &nIsDir);
+    if (dwError) {
+        pr_err("(%s): Error while operating on '%s', '%s'\n",
+                __func__, pszPath, strerror(errno));
+    }
```

Comment:
We can instead use a macro.

I actually use this occasionally for debug builds:
```
$ diff include/tdnf-common-defines.h include/tdnf-common-defines.h.trace
18a19,20
> #define DO_TRACE 1
>
22a25
>             if (DO_TRACE) printf("error %d in %s at %s:%d\n", dwError, __FUNCTION__, __FILE__, __LINE__); \
```
We could add a build switch that enables this debugging code, and we will see a trace for every  `BAIL_ON_TDNF_ERROR` , including function name and line number.

Reply:
This is a good macro while experimenting. Thanks. But this will make user experience difficult.
The two or three places where I added the error message are places which fail with no specific info. There could be more but I found these at the moment. Being a tdnf active user, I got confused and had no idea what the error message meant, so decided to add them.

Reply:
Curious, what *was* the issue? I am okay with adding an error message, but please without the function name and line number. We do not have that anywhere else.

Also for the other change below. The other error messages look okay to me.